### DISCUSSION TITLE
Losing connection to fusionsolar?

### DISCUSSION THREAD

**Discussion by cmsrocon** [wlcrs/huawei_solar#191] [General]
Hi, I've been using the integration for a few days now and I've noticed that every morning the FusionSolar app is showing error 99999999 which indicates an "abnormal connection between management system and equipment". When logging into fusion solar, I can see the inverter loses connectivity in the middle of the night. When I power cycle the inverter, it regains connectivity straight away, so I dont think its a wifi signal issue.

I'm connecting via DD-WRT to the internal wifi AP for the home assistant integration, has anyone else experienced any side effects to connectivity via the home wifi back to huawei?

**Comment by stefanhapper** [comment]
Hello @cmsrocon, I have had the very same issue every time since I am connecting to the inverter via the internal WIFI access point.

@inigoml seems to have found out that if you decrease the query interval to 150 seconds or more the issue doesn't seem to happen. See https://github.com/wlcrs/huawei_solar/issues/48

But I haven't tried this yet - also because the whole point of connecting locally would be for me to have (close to) real-time data.

But it is clearly linked to the Modbus TCP connection from HA. Because when you shut down the HA instance, the connection to Huawei FusionSolar immediately works again.

I wonder if this can be solved by fetching the data via a wired Modbus -> USB connection. Or should we hope for a Firmware update? Has anybody raised this with Huawei yet

@wlcrs any recommendation from your side on this?

**Reply by cmsrocon** [reply]
I've raised the issue with Huawei, but I've not mentioned the HA integration as I'm not sure what the official position is on polling the inverter for data - I'm worried they might remove the ability to pull the data completely? 

They are suggesting its a wifi signal issue, which of course it isn't. Thanks for the info - good to know its a common issue.

**Reply by wlcrs** [reply]
No suggestions from my side, sorry. In the end I'm just another Huawei solar installation owner without much insider knowledge 😅

**Reply by cmsrocon** [reply]
I raised a ticket with huawei who have remotely updated my inverter to SPC125. This morning for the first time I hadn't lost connectivity to fusionsolar. May just be luck - but hoping this might have solved the issue

**Reply by stefanhapper** [reply]
> I raised a ticket with huawei who have remotely updated my inverter to SPC125. This morning for the first time I hadn't lost connectivity to fusionsolar. May just be luck - but hoping this might have solved the issue

Just to let you know that also for me, magically, the Fusion Solar app works again, without any changes. I am still polling the inverter via the access point wifi at an interval of 10 seconds. And I am still on SPC124 firmware.

**Reply by inigoml** [reply]
Just upgraded to SPC125 from 124. Will test. I don't have big expectations about it.
Let's hope if it fixes network HA connection (TCP based).
